src/hooks/useZoom.ts

```diff
@@ -0,0 +1,80 @@
+import { AnimationOptions as OlViewAnimationOptions } from 'ol/View';
+import {
+  useEffect
+} from 'react';
+
+import useMap from './useMap';
+
+/**
+ * This hook adds an interaction to the map and removes/updates it if the dependency array changes.
+ * It accepts an optional active parameter that toggles the active state of the interaction. If it is undefined the
+ * active state will not get changed.
+ * @param constructor returns an interaction to be added to the map, will be called again, if the interaction needs
+ * to be updated
+ * @param active
+ */
+export const useZoom = (
+  /**
+   * Whether the zoom in shall be animated.
+   */
+  animate: boolean,
```

Comment:
Would a custom type e.g. `UseZoomProperties` having default values make sense here?

Reply:
Yes, extracted it to a `MapState` type and changed it to an object.